mono/metadata/threadpool-ms.c

```diff
@@ -728,7 +729,9 @@ worker_try_create (void)
 		counter._.active ++;
 	});
 
-	if ((thread = mono_thread_create_internal (mono_get_root_domain (), worker_thread, NULL, TRUE, 0)) != NULL) {
+	thread = mono_thread_create_internal (mono_get_root_domain (), worker_thread, NULL, TRUE, 0, &error);
+
+	if (thread && mono_error_ok (&error)) {
```

Comment:
This is a code smell, there must be only one way to check for an ok result.

Besides that, there must be a compiling reason for a function to return NULL and not be an error.

By not sorting out those issues, this commit makes things worse.